### PR TITLE
integrate Nuked OPL3 emulator

### PR DESCRIPTION
**Pull request by loki666**

### CODE REVIEW

**Comment by Malvineous** [comment]
Looking good, just a couple of issues:

* Could you avoid changing the default player until everyone has had a chance to test it first?  This could be the source of a lot of complaints if there's a bug we don't notice, so it's good to test first before changing defaults.
* Could you implement surround mode?  Only Ken's synth doesn't work in surround because it uses global variables, and a modern synth should not do this so there should be no problem here.
* Can you avoid the whitespace changes that are unrelated to the code?  If you look at the diff you can see that some lines have been changed that don't have anything to do with the patch.
* Can you try to match the syntax formatting of the surrounding code?  Some of the new lines have different indents and coding style to the existing code which makes it harder to read.  Always best to be consistent with existing code to improve readability.

**Comment by loki666** [comment]
Ok.

For surround mode, the problem is that Nuked OPL3 works only in stereo.
It would require to mix the channels of the buffer, I wasn't sure if that's what we want performance wise.
I can also implement 16bit -> 8bit buffer conversion.

**Comment by Malvineous** [comment]
Ah I see.  In that case it may be best not to worry about the surround OPL for the time being.  Perhaps in future we can investigate something like using the left channel from one instance and the right channel for the other.  The `CSurroundopl` only supports OPL2 anyway so when the time comes it will probably need some changes too.  I'm happy for you to leave out surround OPL support now.  But please fix the typo in the error message in that case :-)

Personally I wouldn't worry about the conversion to 8-bit.

File: src/adplay.cc [hunk 1]
```diff
@@ -437,28 +437,21 @@ int main(int argc, char **argv)
...
-         fprintf(stderr, "Sorry, Nuked OPL3 emulator only works in stereo 16 bits. "
-           "Use --stereo and --16bit options.\n");
-         if(userdb) free(userdb);
-         exit(EXIT_FAILURE);
-       } else {
-         opl = new CNemuopl(cfg.freq);
-       }
-    }
+  	if (cfg.harmonic) {
+  		fprintf(stderr, "Nuked OPL3 emulator doesn't wroks in Surround mode. "
```
Comment (Malvineous) [inline src/adplay.cc]: `wroks` should be `work`

**Comment by Malvineous** [comment]
Many thanks!  Merged in a8d8bc99bc8097aa085a6dad3d46b645d2fa3542

**Pull request closed without merging**